providers/terraform/resources/hcl.go

```diff
@@ -172,24 +172,16 @@ func newMqlHclBlock(runtime *plugin.Runtime, block *hcl.Block, file *hcl.File) (
 
 	snippet := extractHclCodeSnippet(file, block.TypeRange)
 
-	r, err := CreateResource(runtime, "terraform.block", map[string]*llx.RawData{
+	res, err := CreateResource(runtime, "terraform.block", map[string]*llx.RawData{
 		"type":    llx.StringData(block.Type),
 		"labels":  llx.ArrayData(llx.TArr2Raw(block.Labels), types.String),
 		"start":   llx.ResourceData(start, "terraform.fileposition"),
 		"end":     llx.ResourceData(end, "terraform.fileposition"),
 		"snippet": llx.StringData(snippet),
 	})
-
-	/*
-		if err == nil {
-			r.MqlResource().Cache.Store("_hclblock", &resources.CacheEntry{
-				Data: block,
-			})
-			r.MqlResource().Cache.Store("_hclfile", &resources.CacheEntry{
-				Data: file,
-			})
-		}
-	*/
+	r := res.(*mqlTerraformBlock)
+	r.block = plugin.TValue[*hcl.Block]{State: plugin.StateIsSet, Data: block}
```

Comment:
can't we pass this in the args as an mql resource, i.e. `llx.ResourceData`?

Reply:
this is for the internal implementation, 

```

type mqlTerraformBlockInternal struct {
	block      plugin.TValue[*hcl.Block]
	cachedFile plugin.TValue[*hcl.File]
}
```